src/triangular.jl

```diff
@@ -1,11 +1,19 @@
-@inline transpose(A::LinearAlgebra.LowerTriangular{<:Any,<:StaticMatrix}) =
-    LinearAlgebra.UpperTriangular(transpose(A.data))
-@inline adjoint(A::LinearAlgebra.LowerTriangular{<:Any,<:StaticMatrix}) =
-    LinearAlgebra.UpperTriangular(adjoint(A.data))
-@inline transpose(A::LinearAlgebra.UpperTriangular{<:Any,<:StaticMatrix}) =
-    LinearAlgebra.LowerTriangular(transpose(A.data))
-@inline adjoint(A::LinearAlgebra.UpperTriangular{<:Any,<:StaticMatrix}) =
-    LinearAlgebra.LowerTriangular(adjoint(A.data))
+@inline transpose(A::LowerTriangular{<:Any,<:StaticMatrix}) =
+    UpperTriangular(transpose(A.data))
+@inline adjoint(A::LowerTriangular{<:Any,<:StaticMatrix}) =
+    UpperTriangular(adjoint(A.data))
+@inline transpose(A::UnitLowerTriangular{<:Any,<:StaticMatrix}) =
+    UnitUpperTriangular(transpose(A.data))
+@inline adjoint(A::UnitLowerTriangular{<:Any,<:StaticMatrix}) =
+    UnitUpperTriangular(adjoint(A.data))
+@inline transpose(A::UpperTriangular{<:Any,<:StaticMatrix}) =
+    LowerTriangular(transpose(A.data))
+@inline adjoint(A::UpperTriangular{<:Any,<:StaticMatrix}) =
+    LowerTriangular(adjoint(A.data))
+@inline transpose(A::UnitUpperTriangular{<:Any,<:StaticMatrix}) =
+    UnitLowerTriangular(transpose(A.data))
+@inline adjoint(A::UnitUpperTriangular{<:Any,<:StaticMatrix}) =
+    UnitLowerTriangular(adjoint(A.data))
```

Comment:
It's surprising that Base doesn't seem to include these rules.

Reply:
Is it worth opening an issue in Julia repository?

Reply:
Sure. I can't think of a reason these shouldn't be structure-preserving.